pages/glossary/visible-changes-in-content.md

```diff
@@ -0,0 +1,22 @@
+---
+title: Visible changes in content
+key: visible-changes-in-content
+unambiguous: true
+objective: true
+input_aspects:
+  - CSS styling
+  - DOM tree
+---
+
+A _[event][] originated change in the content_ of a [web page][] occurs when, by comparing the [web page][] before and after the event [firing][] the rendered pixels change in any part of the document that is currently within the [viewport][] or that can be brought into the [viewport][] via scrolling.
+
+- If the [web page][] is rendering time-based media, rendered pixels and audio will be changing as part of the playback. The comparison in this instance should compare the pixels and audio that are rendered if the event is not fired, with the ones that are rendered if the event is fired.
+
+**Assumptions:**
+
+- This definition assumes that there are no changes in the content of the [web page][] caused by another [event][]. If this is not the case, changes may be attributed to the wrong event.
```

Comment:
This is a real issue that I would be a little wary of assuming doesn't occur. Especially in the context of automated testing, correlating events to content changes is extremely tricky as you'll inevitably have to deal with interference from non-event based content changes

Reply:
I agree, but I couldn't come up with a way that would unequivocally identify the changes that are triggered by an event when there are multiple events causing changes. Would love to get a suggestion if you have one :-)

And in the scope of WAI-Tools were saying that this is a manual rule, so I assume that this assumption was easier to write in this context...

Reply:
I've experimented with correlating events and mutation changes using frame renders as a heuristic in the past, and it worked reasonably well. That did lead me to this realisation though: https://github.com/act-rules/act-rules.github.io/pull/1396#pullrequestreview-492408701